### PR TITLE
Improve install process

### PR DESCRIPTION
**Pull request by olalonde**
I dockerized the app so that it can be easily installed as a Deis app. The Deis API is now configurable through environment variable.

``` bash
deis create
deis config:set DEIS_API=http://deis.local3.deisapp.com
git push deis master
```

### CODE REVIEW

**Comment by jumbojett** [comment]
Thank you @olalonde

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM node:0.10
```
Comment (Joshua-Anderson) [inline Dockerfile:1]: Any reason we are using node v0.10 and not v0.12?

**Comment by Joshua-Anderson** [comment]
I like the idea of this but I think that the web site should be built outside of docker and all that should be in the container is the compiled website and a webserver like nginx.

**Comment by Joshua-Anderson** [comment]
Also @jumbojett should publish this to docker hub so we can `deis pull` this into deis :)

File: Dockerfile (hunk 1 above)
Comment (olalonde) [inline Dockerfile:1]: Nope, I'll fix this.

**Comment by bacongobbler** [comment]
> should be in the container is the compiled website and a webserver like nginx.

:+1:

**Comment by bacongobbler** [comment]
FWIW this is what we once attempted to do with the docs; just haven't had time to work out the kinks on how the build process looks like

**Comment by jumbojett** [comment]
@Joshua-Anderson I like the way you think. :+1: 

`docker pull jumbojett/deis-ui`

https://registry.hub.docker.com/u/jumbojett/deis-ui/

**Comment by olalonde** [comment]
> I like the idea of this but I think that the web site should be built outside of docker and all that should be in the container is the compiled website and a webserver like nginx.

Why? Seems like a text book use case for Docker. 

Regarding the server, nginx would definitely be better than `grunt serve:dist` and it would be relatively easy to do if the deis build process has access to the app's environment variables (those set with `deis config:set`).

**Comment by bacongobbler** [comment]
> Why? Seems like a text book use case for Docker.

To simplify, deploying a 10MB container with the compiled HTML + embedded nginx binary has less surface area and is significantly smaller than `node:0.10`. We do the exact same thing as well in Deis... Nobody wants to wait for 1GB+ components based on `golang:1.3.1` to deploy. The final binary size for a compiled `deis/logger` sits just over 5MB (this was even more accented when DockerHub was in its early stages and downtime or network latency occurred _quite_ frequently).

here's a great blog post on the same discussion: http://livewyer.com/blog/2015/02/24/slimming-down-your-docker-containers-alpine-linux

also some of the same principles in Deis: https://github.com/deis/deis/blob/master/builder/Makefile#L16-L21

**Comment by olalonde** [comment]
Would that mean maintaining keeping the built files under version control so that I can `git push` deploy to deis? Or is there another way to do it? Another option would be to `rm -rf` everything that is not needed after the build steps in the Dockerfile. Personally as an end-user, all I care about is how fast I can install deis-ui (I spent >1 hour to get it running with current README instructions though this was my first Deis deploy).

**Comment by Joshua-Anderson** [comment]
You can do a `deis pull` to pull the image from the docker registry, so you don't have to build it.

**Pull request merged**